src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Providers/Wrapper/Constants.cs

```diff
@@ -82,6 +83,24 @@ public enum DatastoreVendor
         Other
     }
 
+    public static class DatastoreVendorExtensions
+    {
+        // Convert our internal enum to the matching OTel "known" name for a database provider
+        public static string ToKnownName(this DatastoreVendor vendor)
+        {
+            switch (vendor)
+            {
+                case DatastoreVendor.Other:
+                    return "other_sql";
+                case DatastoreVendor.IBMDB2:
+                    return "db2";
+                // The others match our enum name
+                default:
+                    return vendor.ToString().ToLower();
```

Comment:
Calling ToString on an enum involves reflection. I think we already have code that can be used to cache the results so that we don't need to incur the reflection cost each time. We should probably leverage that code here.

Reply:
Ah I wasn't aware of `EnumNameCache`. I can do that.

Reply:
It's literally the next line in that file. 🤦